contracts/dao/TribeMinter.sol

```diff
@@ -0,0 +1,211 @@
+        // Deplete buffer by the newly increased cap difference
+        if (newBufferCap > oldBufferCap) {
+            uint256 increment = newBufferCap - oldBufferCap;
+            _depleteBuffer(increment);
+        }
+    }
+
+    /// @dev no-op, reverts. Prevent admin or governor from overwriting ideal rate limit
+    function setRateLimitPerSecond(uint256) external pure override {
+        revert("no-op");
+    }
+
+    /// @dev no-op, reverts. Prevent admin or governor from overwriting ideal buffer cap
+    function setBufferCap(uint256) external pure override { 
+        revert("no-op");
+    }
+
+    /// @notice mints TRIBE to the target address, subject to rate limit
+    /// @param to the address to send TRIBE to
+    /// @param amount the amount of TRIBE to send
+    function mint(address to, uint256 amount) external override onlyGovernorOrAdmin {
+        // first apply rate limit 
+        _depleteBuffer(amount);
+
+        // then mint
+        _mint(to, amount);
+    }
+
+    /// @notice add an address to the lockedTribe excluded list
+    function addLockedTribeAddress(address lockedTribeAddress) external override onlyGovernorOrAdmin {
+        _addLockedTribeAddress(lockedTribeAddress);
+    }
+
+    /// @notice remove an address from the lockedTribe excluded list
+    function removeLockedTribeAddress(address lockedTribeAddress) external override onlyGovernorOrAdmin {
+        _lockedTribeAddresses.remove(lockedTribeAddress);
+        emit RemoveLockedTribeAddress(lockedTribeAddress);
+    }
+
+    /// @notice changes the TRIBE minter address
+    /// @param newMinter the new minter address
+    function setMinter(address newMinter) external override onlyOwner {
+        require(newMinter != address(0), "TribeReserveStabilizer: zero address");
+        ITribe _tribe = ITribe(address(tribe()));
+        _tribe.setMinter(newMinter);
+    }
+
+    /// @notice sets the max annual inflation relative to current supply
+    /// @param newAnnualMaxInflationBasisPoints the new max inflation % denominated in basis points (1/10000)
+    function setAnnualMaxInflationBasisPoints(uint256 newAnnualMaxInflationBasisPoints) external override onlyGovernorOrAdmin {
+        _setAnnualMaxInflationBasisPoints(newAnnualMaxInflationBasisPoints);
+    }
+
+    /// @notice return the ideal buffer cap based on TRIBE circulating supply
+    function idealBufferCap() public view override returns (uint256) {
+        return tribeCirculatingSupply() * annualMaxInflationBasisPoints / Constants.BASIS_POINTS_GRANULARITY;
+    }
+
+    /// @notice return the TRIBE supply, subtracting locked TRIBE
+    function tribeCirculatingSupply() public view override returns (uint256) {
+        IERC20 _tribe = tribe();
+
+        // Remove all locked TRIBE from total supply calculation
+        uint256 lockedTribe;
+        for (uint256 i = 0; i < _lockedTribeAddresses.length(); i++) {
+            lockedTribe += _tribe.balanceOf(_lockedTribeAddresses.at(i));
```

Comment:
Using the literal tribe balance of the locked address adds in a potential current (and if not current, potential future) exploit: atomically flashloaning a TON of Tribe into a locked-tribe-address, calling this function, and using the atomically-reduced-circulating-tribe-supply to inflate greater than intended.

you might say "if the tribe goes to the locked address it's locked!" - but that is not necessarily the case. if we keep the implementation this way we will need to carefully consider every single locked address we add to ensure that there isn't any functionality for recovering tokens accidentally sent to that address, and we'll need to remember this for every future locked address we add

also - consider the scenario in which tokens vest from a timelock. those addresses contain *some* locked and *some* circulating tribe, and allow the owners of the vested-portion to manipulate the Tribe inflation rate this way if they wish. this seems pretty problematic.